agent-core/src/main/java/com/newrelic/agent/android/instrumentation/okhttp3/OkHttp3Instrumentation.java

```diff
@@ -58,6 +63,17 @@ public static Call newCall(OkHttpClient client, Request request) {
         return new CallExtension(client, request, client.newCall(request), transactionState);
     }
 
+    private static void addHeadersAsCustomAttribute(TransactionState transactionState, Request request) {
+
+        Map<String, String> headers = new HashMap<>();
+        for (String s : HttpHeaders.getInstance().getHttpHeaders()) {
+            if (request.headers().get(s) != null) {
+                headers.put(Constants.ApolloGraphQLHeader.normalizeApolloHeader(s), request.headers().get(s));
```

Comment:
This is really confusing. The method seems general enough, but uses a `ApolloGraphQLHeader()`* constant static method* to normalize it? 

You can also call the `request.headers.contains(s)` method as well and avoid null checks.